_posts/2019-04-30-accessing-kafka-part-3.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: post
+title:  "Accessing Kafka: Part 3 - OpenShift Routes"
+date: 2019-04-30
+author: jakub_scholz
+---
+
+In the third part of this blog post series we will look at exposing Kafka using OpenShift Routes.
+This post will explain how routes work and how they can be used with Kafka.
+Routes are available only on OpenShift.
+But if you are a Kubernetes user, don't be sad, one of the next part will be about using Kubernetes Ingress which is similar to OpenShift routes.
+
+<!--more-->
+
+_This post is part of a bigger series about different ways to access a Kafka cluster powered by Strimzi.
+The other parts published so far are:_
+
+* _[Part 1 - Introduction](https://strimzi.io/2019/04/17/accessing-kafka-part-1.html)_
+* _[Part 2 - Node Ports](https://strimzi.io/2019/04/23/accessing-kafka-part-2.html)_
+* _Part 3 - OpenShift Routes (this post)_
+
+# OpenShift Routes
+
+Routes are an OpenShift concept for exposing services to the outside of the OpenShift platform.
+Routes handle both data routing as well as DNS resolution.
+DNS resolution is usually handled using [wildcard DNS entries](https://en.wikipedia.org/wiki/Wildcard_DNS_record).
+That allows OpenShift to assign each route its own DNS name which is based on the wildcard entry.
+Users do not have to anything special to handle the DNS records.
+But don't worry, when you don't own any domains where you can setup the wildcard entires, it can use services such as [nip.ip](https://nip.io/) for the wildcard DNS routing.
+Data routing is done using the [HAProxy](https://www.haproxy.org) load balancer which serves as the router behind the routes.
+
+The main use-case of the router is HTTP(S) routing.
+The routes are able to do path based routing of HTTP and HTTPS (with TLS termination) traffic.
+In this mode, the HTTP requests will be routed to different services based on the request path.
+However, since the Kafka protocol is not based on HTTP, the HTTP features are not very useful for Strimzi and Kafka brokers.
+
+But luckily the routes can be also used for TLS passthrough.
+In this mode, it uses TLS SNI to determine the service to which the traffic should be routed and passes the TLS connection to the service (and eventually to the pod backing the service) without decoding it.
+This mode is what Strimzi is using for exposing Kafka.
+
+If you want to learn more about OpenShift Routes, check the [OpenShift documentation](https://docs.openshift.com/container-platform/3.11/architecture/networking/routes.html).
+
+# Exposing Kafka using OpenShift Routes
+
+Exposing Kafka using OpenShift Routes is probably the easiest of all the available options.
+All you need to do is to configure it in the Kafka custom resource.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    listeners:
+      # ...
+      external:
+        type: route
+    # ...
+```
+
+And Strimzi Kafka Operator and OpenShift will take care of the rest.
```

Comment:
```suggestion
And the Strimzi Cluster Operator and OpenShift will take care of the rest.
```

Reply:
_Strimzi Kafka operator_ is there intentionally because of the SEO ;-)